app/src/main/java/com/group80/uoftinder/ContactsActivity.java

```diff
@@ -0,0 +1,134 @@
+import com.group80.uoftinder.firebase.storage.ImageStorageDbFacade;
+import com.group80.uoftinder.firebase.storage.StorageDbDownloadable;
+
+/**
+ * A window that displays all the contacts to chat with
+ */
+public class ContactsActivity extends AppCompatActivity {
+    FirestoreRecyclerAdapter<ContactModel, ContactViewHolder> contactAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_contacts);
+
+        // TODO: remove after having functioning log-in ********************************************
+        FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+        firebaseAuth.signInWithEmailAndPassword("csc207.group80.uoftinder@gmail.com", "CSC207Group80!").addOnFailureListener(Throwable::printStackTrace);
+//        firebaseAuth.signInWithEmailAndPassword("csc207.group80.uoftinder.bot@gmail.com", "12345678").addOnFailureListener(Throwable::printStackTrace);
+
+        // TODO: wrap-up FirebaseFirestore
+        FirebaseFirestore firebaseFirestore = FirebaseFirestore.getInstance();
+        String uid = firebaseAuth.getCurrentUser().getUid();
+        Query query = firebaseFirestore.collection("Users").whereArrayContains("contacts", uid);
+        //******************************************************************************************
+
+        FirestoreRecyclerOptions<ContactModel> contacts = new FirestoreRecyclerOptions.Builder<ContactModel>().setQuery(query, ContactModel.class).build();
+        contactAdapter = new FirestoreRecyclerAdapter<ContactModel, ContactViewHolder>(contacts) {
+            @Override
+            protected void onBindViewHolder(@NonNull ContactViewHolder holder, int position, @NonNull ContactModel contactModel) {
+                holder.onBind(getApplicationContext(), position, contactModel);
+            }
+
+            @NonNull
+            @Override
+            public ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+                View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.uoftinder_contact_list_contact_layout, parent, false);
+                return new ContactViewHolder(view);
+            }
+        };
+
+        RecyclerView recyclerView = findViewById(R.id.contactActivityRecyclerView);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(contactAdapter);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        contactAdapter.startListening();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (contactAdapter != null) {
+            contactAdapter.stopListening();
+        }
+    }
+
+    private class ContactViewHolder extends RecyclerView.ViewHolder {
```

Comment:
It's usually bad practice to put two classes inside the same file. I would move this to a file called ContactViewHolder.java

Reply:
Indeed in most cases defining two classes in one file would be a bad practice, but nested class is a little different (later on that) and has its own pros and cons, and currently, this is a workaround for the current uncertainty of our package structure (since I can only make a class `public` or package-private). 

After everything works and during the final refactoring stage, I will extract `ContactViewHolder` into a separate file, give it no modifier so it is package-private, and adjust the package structure so only `ContactsActivity` is in the same package with it to avoid this class being accessed elsewhere. 

About nested classes, see [When to Use Nested Classes, Local Classes, Anonymous Classes, and Lambda Expressions](https://docs.oracle.com/javase/tutorial/java/javaOO/whentouse.html) and [Nested Classes](https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html) from Oracle.

Reply:
Ok, sounds good!